### PR TITLE
COMPATIBILITY: Modernize the component 

### PR DESCRIPTION
**Pull request by Arkshine**
This PR modernizes the components:
* Use the new header API
* Convert all the widgets to glimmer components
* Fix a few UX issues
* Add acceptance tests

The HTML structure and CSS classes should be the same.

https://github.com/paviliondev/discourse-dropdown-header/assets/360640/4915f577-e3b1-4c0c-8f63-ea93460ab22e

### CODE REVIEW

File: common/common.scss [hunk 1]
```diff
@@ -81,7 +81,7 @@
 }
 
 // hide on scroll when nav is positioned to left of header:
-.title .custom-header-links.scrolling {
+.d-header .custom-header-links.scrolling {
```
Comment (Arkshine) [inline common/common.scss:94]: The plugin outlet wrapper is not inside `.title`.

**Comment by angusmcleod** [comment]
@Arkshine Amazing stuff!

**Comment by angusmcleod** [comment]
@nolosb Would you mind reviewing this one?

**Comment by merefield** [comment]
@Arkshine checks are failing currently?

**Comment by Arkshine** [comment]
> @Arkshine checks are failing currently?

~~I tried to update the CI linting workflow. I'm unsure if it will work.~~
~~Linting requires at least node >= 18, and I added the missing `gjs` and `hbs` extensions to eslint and prettier.~~

I think we can use the Discourse theme reusable workflow here. It refers to https://github.com/discourse/.github/blob/main/.github/workflows/discourse-theme.yml.

**Review by Arkshine — COMMENTED** [review]
This fixes the "_Interaction added to non-interactive element_" error.
https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-invalid-interactive.md

Ideally, I think we want to use `<a>` link here for better accessibility. 
That's something I will see to change in another PR.

File: javascripts/discourse/components/custom-header-dropdown.gjs [hunk 2]
```diff
@@ -25,6 +25,7 @@ export default class CustomHeaderDropdown extends Component {
     <li
       class="custom-header-dropdown-link"
       title={{@item.title}}
+      role="button"
```
Comment (Arkshine) [inline javascripts/discourse/components/custom-header-dropdown.gjs:28]: This fixes the "_Interaction added to non-interactive element_" error.
https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-invalid-interactive.md

Ideally, I think we want to use `<a>` link here for better accessibility. 
That's something I will see to change in another PR.

**Comment by angusmcleod** [comment]
@nolosb Is currently unavailable, however Github is suggesting @nathan-nz review this :) 

Screenshot 2024-07-10 at 15 48 19

What do you think @nathan-nz?

**Comment by nathan-nz** [comment]
> @nolosb Is currently unavailable, however Github is suggesting @nathan-nz review this :)
> What do you think @nathan-nz?

I'm no dev (as you know!) so it won't be an expert review. But I am very happy to approve it as it has had your and @merefield's eyes on it (and @Arkshine is trustworthy).

**Review by nathan-nz — APPROVED** [review]
Thanks for your excellent work @Arkshine

**Comment by nathan-nz** [comment]
@Arkshine - I'm not sure what this means:
image
Is this something you can address?

**Comment by Arkshine** [comment]
@nathan-nz Yes, I've removed the workflow. Since we are using the Discourse reusable workflow, and the test part is already included, it was a duplicate. https://github.com/discourse/.github/blob/main/.github/workflows/discourse-theme.yml#L109

**Pull request merged**